mixer/adapter/bypass/bypass.go

```diff
@@ -0,0 +1,302 @@
+	conn                *grpc.ClientConn
+	infraClient         v1beta1.InfrastructureBackendClient
+	checknothingClient  checknothing.HandleCheckNothingServiceClient
+	reportnothingClient reportnothing.HandleReportNothingServiceClient
+	metricClient        metric.HandleMetricServiceClient
+	quotaClient         quota.HandleQuotaServiceClient
+}
+
+var _ checknothing.Handler = &handler{}
+var _ reportnothing.Handler = &handler{}
+var _ metric.Handler = &handler{}
+var _ quota.Handler = &handler{}
+
+func (h *handler) HandleCheckNothing(ctx context.Context, instance *checknothing.Instance) (adapter.CheckResult, error) {
+	request := checknothing.HandleCheckNothingRequest{}
+
+	request.DedupId = newDedupID()
+	request.AdapterConfig = h.config
+
+	request.Instance = &checknothing.InstanceMsg{
+		Name: instance.Name,
+	}
+
+	result := adapter.CheckResult{}
+	response, err := h.checknothingClient.HandleCheckNothing(ctx, &request)
+	if err == nil {
+		result.Status = response.Status
+		result.ValidDuration = response.ValidDuration
+		result.ValidUseCount = response.ValidUseCount
+	}
+
+	return result, err
+}
+
+func (h *handler) HandleReportNothing(ctx context.Context, instances []*reportnothing.Instance) error {
+	request := reportnothing.HandleReportNothingRequest{}
+
+	request.DedupId = newDedupID()
+	request.AdapterConfig = h.config
+
+	for _, ins := range instances {
+		instance := &reportnothing.InstanceMsg{
+			Name: ins.Name,
+		}
+
+		request.Instances = append(request.Instances, instance)
+	}
+
+	_, err := h.reportnothingClient.HandleReportNothing(ctx, &request)
+
+	return err
+}
+
+func (h *handler) HandleMetric(ctx context.Context, instances []*metric.Instance) error {
+	request := metric.HandleMetricRequest{}
+
+	request.DedupId = newDedupID()
+	request.AdapterConfig = h.config
+
+	for _, ins := range instances {
+		value, err := convertValue(ins.Value)
+		if err != nil {
+			return err
+		}
+		dimensions, err := convertMapValue(ins.Dimensions)
+		if err != nil {
+			return err
+		}
+		instance := &metric.InstanceMsg{
+			Name:       ins.Name,
+			Value:      value,
+			Dimensions: dimensions,
+		}
+
+		request.Instances = append(request.Instances, instance)
+	}
+
+	_, err := h.metricClient.HandleMetric(ctx, &request)
+
+	return err
+}
+
+func (h *handler) HandleQuota(ctx context.Context, instance *quota.Instance, args adapter.QuotaArgs) (adapter.QuotaResult, error) {
+	request := quota.HandleQuotaRequest{}
+
+	request.DedupId = newDedupID()
+	request.AdapterConfig = h.config
+
+	result := adapter.QuotaResult{}
+
+	dimensions, err := convertMapValue(instance.Dimensions)
+	if err != nil {
+		return result, err
+	}
+
+	request.Instance = &quota.InstanceMsg{
+		Name:       instance.Name,
+		Dimensions: dimensions,
+	}
+
+	response, err := h.quotaClient.HandleQuota(ctx, &request)
+	if err == nil {
+		result.ValidDuration = response.Quotas[instance.Name].ValidDuration
+		result.Amount = response.Quotas[instance.Name].GrantedAmount
```

Comment:
response also has quotas, do we need to map that too ?
https://github.com/istio/api/blob/3c1a4e62eb905aee4fb06f39e044081898c1dab1/mixer/adapter/model/v1beta1/quota.proto#L56